File: src/53. Maximum Subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */

export function maxSubArray(nums) {
  let curMax = 0,
    sumMax = -Number.MAX_VALUE
  for (let val of nums) {
    curMax = val + Math.max(curMax, 0)
    sumMax = Math.max(sumMax, curMax)
  }
  return sumMax
}

File: src/53. Maximum Subarray.cpp
class Solution {
    public:
        int maxSubArray(vector < int > & nums) {
            int maxSum = INT_MIN;
            int localMaxSum = 0;
            for (int num: nums) {
                localMaxSum = max(localMaxSum + num, num);
                maxSum = max(maxSum, localMaxSum);
            }
            return maxSum;
        }
};